docs/src/main/asciidoc/bigquery.adoc

```diff
@@ -54,13 +59,98 @@ public void runQuery() throws InterruptedException {
+  BigQueryFileMessageHandler messageHandler = new BigQueryFileMessageHandler(bigQueryTemplate);
+  messageHandler.setFormatOptions(FormatOptions.csv());
+  messageHandler.setOutputChannel(bigQueryJobReplyChannel());
+  return messageHandler;
+}
+----
+
+==== BigQuery Message Handling
+
+The `BigQueryFileMessageHandler` accepts the following message payload types for loading into BigQuery: `java.io.File`, `byte[]`, `org.springframework.core.io.Resource`, and `java.io.InputStream`.
+The message payload will be streamed and written to the BigQuery table you specify.
+
+By default, the `BigQueryFileMessageHandler` is configured to read the headers of the messages it receives to determine how to load the data.
+The headers are specified by the class `BigQuerySpringMessageHeaders` and summarized below.
+
+|=========================================
+| Header | Description
+| `BigQuerySpringMessageHeaders.TABLE_NAME` | Specifies the BigQuery table within your dataset to write to.
+| `BigQuerySpringMessageHeaders.FORMAT_OPTIONS` | Describes the data format of your data to load (i.e. CSV, JSON, etc.).
+|=========================================
+
+Alternatively, you may omit these headers and explicitly set the table name or format options by calling `setTableName(...)` and `setFormatOptions(...)`.
+
+==== BigQuery Message Reply
+
+After the `BigQueryFileMessageHandler` processes a message to load data to your BigQuery table, it will respond with a `Job` on the reply channel.
+The https://googleapis.dev/java/google-cloud-clients/latest/index.html?com/google/cloud/bigquery/package-summary.html[Job object] provides metadata and information about the load file operation.
+
+By default, the `BigQueryFileMessageHandler` is run in asynchronous mode, with `setSync(false)`, and it will reply with a `ListenableFuture<Job>` on the reply channel.
+The future is tied to the status of the data loading job and will complete when the job completes.
+
+If the handler is run in synchronous mode with `setSync(true)`, then the handler will block on the completion of the loading job and block until it is complete.
+
+NOTE: If you decide to use Spring Integration Gateways and you wish to receive `ListenableFuture<Job>` as a reply object in the Gateway, you will have to call `.setAsyncExecutor(null)` on your `GatewayProxyFactoryBean`.
+This is needed to indicate that you wish to reply on the built-in async support rather than rely on async handling of the gateway.
+
 === Configuration
 
 The following application properties may be configured with Spring Cloud GCP BigQuery libraries.
 
 |===========================================================================
 | Name | Description | Required | Default value
-| `spring.cloud.gcp.bigquery.enabled` | Enables or disables Spring Cloud GCP BigQuery autoconfiguration | No | `true`
+| `spring.cloud.gcp.bigquery.datasetName` | The BigQuery dataset that the `BigQueryTemplate` and `BigQueryFileMessageHandler` is scoped to. | Yes |
```

Comment:
This comes with sample, right?

Reply:
Yessir, the sample PR is out for review.